executor/runtime/docker/docker.go

```diff
@@ -216,6 +216,7 @@ type DockerRuntime struct { // nolint: golint
 	pidCgroupPath     string
 	cfg               config.Config
 	dockerCfg         Config
+	hostCfg           *container.HostConfig
```

Comment:
Since we only use our runtimes "once", does it make sense to change the API for the runtimes to look closer to that?

Reply:
I'm not sure what you mean - are you saying store the runtime here instead of HostConfig, or something else?

Reply:
OK - doing this breaks the contract for `type Runtime interface`: previously, you could (in theory), call `Prepare()` and `Start()` for more than one container.  Adding this dependency breaks that contract.

I changed the code to add the runtime to the `Container` struct instead, which preserves this behaviour - we might not use it currently, but I don't have a pressing need to break this contract right now, so I might as well keep it as is.